src/System.Security.Cryptography.X509Certificates/tests/ExtensionsTests.cs

```diff
@@ -213,39 +214,56 @@ public static void BasicConstraintsExtensionDefault()
             Assert.Equal(0, e.PathLengthConstraint);
         }
 
-        [Fact]
+        [Theory]
+        [MemberData("BasicConstraintsData")]
         [ActiveIssue(1993, PlatformID.AnyUnix)]
-        public static void BasicConstraintsExtension_Leaf()
+        public static void BasicConstraintsExtensionEncode(
+            bool certificateAuthority,
+            bool hasPathLengthConstraint,
+            int pathLengthConstraint,
+            bool critical,
+            string expectedDerString)
         {
-            TestBasicConstraintsExtension(false, false, 0, false, "3000".HexToByteArray());
-        }
+            X509BasicConstraintsExtension ext = new X509BasicConstraintsExtension(
+                certificateAuthority,
+                hasPathLengthConstraint,
+                pathLengthConstraint,
+                critical);
 
-        [Fact]
-        [ActiveIssue(1993, PlatformID.AnyUnix)]
-        public static void BasicConstraintsExtension_CA_NoLength()
-        {
-            TestBasicConstraintsExtension(true, false, 0, false, "30030101ff".HexToByteArray());
+            byte[] expectedDer = expectedDerString.HexToByteArray();
+            Assert.Equal(expectedDer, ext.RawData);
         }
 
-        [Fact]
-        [ActiveIssue(1993, PlatformID.AnyUnix)]
-        public static void BasicConstraintsExtension_Leaf_Length0()
+        [Theory]
+        [MemberData("BasicConstraintsData")]
+        public static void BasicConstraintsExtensionDecode(
+            bool certificateAuthority,
+            bool hasPathLengthConstraint,
+            int pathLengthConstraint,
+            bool critical,
+            string rawDataString)
         {
-            TestBasicConstraintsExtension(false, true, 0, false, "3003020100".HexToByteArray());
-        }
+            byte[] rawData = rawDataString.HexToByteArray();
 
-        [Fact]
-        [ActiveIssue(1993, PlatformID.AnyUnix)]
-        public static void BasicConstraintsExtension_LeafLongPath()
-        {
-            TestBasicConstraintsExtension(false, true, 7654321, false, "3005020374cbb1".HexToByteArray());
+            X509BasicConstraintsExtension ext = new X509BasicConstraintsExtension(new AsnEncodedData(rawData), critical);
+            Assert.Equal(certificateAuthority, ext.CertificateAuthority);
+            Assert.Equal(hasPathLengthConstraint, ext.HasPathLengthConstraint);
+            Assert.Equal(pathLengthConstraint, ext.PathLengthConstraint);
         }
 
-        [Fact]
-        [ActiveIssue(1993, PlatformID.AnyUnix)]
-        public static void BasicConstraintsExtension_CA_559()
+        public static IEnumerable<object[]> BasicConstraintsData
```

Comment:
I thought about that, but decided to use MemberData so I could reuse the data and didn't have to duplicate it across the two Theories. Do you think it would be leaps and bounds better to use InlineData?